index.js

```diff
@@ -32,41 +30,56 @@ exports.kinesisHandler = (records, opts = {}, context) => {
       });
     }
 
-    // Required parameters are valid execute the following:
-    // 1) Obtain the decoded kinesis data
-    // 2) Obtain a valid OAuth token to process record data
     const schema = opts.schema;
     const apiUri = opts.apiUri;
+    const hrcModel = new HoldRequestConsumerModel();
     const streamsClient = new NyplStreamsClient({ nyplDataApiClientBase: apiUri });
     const apiHelper = new ApiServiceHelper(
       process.env.OAUTH_PROVIDER_URL,
       process.env.OAUTH_CLIENT_ID,
       process.env.OAUTH_CLIENT_SECRET,
       process.env.OAUTH_PROVIDER_SCOPE
     );
-    const holdRequestConsumerModel = new HoldRequestConsumerModel();
 
     Promise.all([
       apiHelper.getOAuthToken(CACHE['access_token']),
       streamsClient.decodeData(schema, records.map(i => i.kinesis.data))
     ]).then(result => {
-      // The access_token has been obtained and all records have been grouped by source
-      // Next, we need create a string from the record and nyplSource keys to perform a GET request
-      // to the ItemService
       CACHE['access_token'] = result[0];
-      // Save the decoded records to the Model Object
-      holdRequestConsumerModel.setRecords(result[1]);
-
-      const groupedRecordsBySource = apiHelper.groupRecordsBy(holdRequestConsumerModel.getRecords(), 'nyplSource');
-      const groupedRecordsWithApiUrl = apiHelper.generateRecordApiUrlsArray(groupedRecordsBySource, apiUri);
-      console.log(groupedRecordsWithApiUrl);
+      hrcModel.setRecords(result[1]);
+      return apiHelper.generateItemApiUrlsArray(hrcModel.getRecords(), apiUri);
+    })
+    .then(itemServiceApiUrlsArray => {
+      hrcModel.setItemServiceApiUrls(itemServiceApiUrlsArray);
+      return apiHelper.processBatchRequest(hrcModel.getItemServiceApiUrls(), CACHE['access_token'], 'itemApi');
```

Comment:
As we discussed, if the token expires at this point, you may have to handle it and decide whether to make the entire lambda fail or just start running from this point on down stream in the promise chain (assuming that is possible)